matcher.go

```diff
@@ -1,43 +1,61 @@
 package gostatsd
 
 import (
+	"regexp"
 	"strings"
 )
 
 type StringMatch struct {
 	test        string
 	invertMatch bool
 	prefixMatch bool
+	isRegex     bool
+	regex       *regexp.Regexp
```

Comment:
This field can be used as a flag instead of the bool above. It's not a regex if the pointer is `nil`.